src/routes/blog/post/integrate-sql-nosql-vector-graph-or-any-database-into-your-appwrite-project/+page.markdoc

```diff
@@ -0,0 +1,340 @@
+await client`INSERT INTO warehouses (location, capacity) VALUES (${location}, ${capacity})`;
+
+// Query data
+const page = 1;
+const limit = 100;
+const warehouses = await client`SELECT * FROM warehouses LIMIT ${limit} OFFSET ${(page - 1) * limit}`;
+return res.json(warehouses);
+```
+
+You can find an entire example in our [query Neon Postgres](https://github.com/appwrite/templates/tree/main/node/query-neon-postgres) function template.
+
+![Neon Postgres Console](/images/blog/integrate-sql-nosql-vector-graph-or-any-database-into-your-appwrite-project/neon.png)
+
+## MongoDB Atlas
+
+Atlas, a managed cloud platform by the MongoDB team, is a cloud provider of MongoDB clusters.
+
+MongoDB database is well-known for its JSON document structure. This approach allows you to store unstructured data optimally, with a huge improvement boost to write operations. For instance, usage stats or sales history could be stored in MongoDB and later processed to gain more insights into app analytics.
+
+While MongoDB shines the most in storing unstructured data, nowadays, it provides all the capabilities developers expect from a relational database. If you don’t like SQL queries, MongoDB is a great choice.
+
+With Mongo, integration becomes simpler, since you don’t need to define structure of collections or documents. After preparing a client, you can directly start writing and querying data.
+
+```jsx
+import { MongoClient, ServerApiVersion } from 'mongodb';
+
+let client;
+
+export default async ({ req, res, log, error }) => {
+  // Connect to database
+  if(!client) {
+    client = new MongoClient(process.env.MONGO_URI, {
+      serverApi: {
+        version: ServerApiVersion.v1,
+        strict: true,
+        deprecationErrors: true,
+      }
+    });
+
+    await client.connect();
+  }
+
+  // Insert data
+  const location = `Street ${Math.round(Math.random() * 1000)}, Earth`; // Random address
+  const capacity = 10 + Math.round(Math.random() * 10) * 10; // Random number: 10,20,30,...,90,100
+
+  await client.db("main").collection("warehouses").insertOne({
+      location,
+      capacity
+  });
+
+  // Query data
+  const page = 1;
+  const limit = 100;
+  const cursor = client.db("main").collection("warehouses")
+    .find().limit(limit).skip((page - 1) * limit);
+
+  const docs = [];
+  for await (const doc of cursor) {
+    docs.push(doc);
+  }
+
+  return res.json(docs);
+};
+```
+
+A better-structured example can be found in the [query Mongo Atlas](https://github.com/appwrite/templates/tree/main/node/query-mongo-atlas) function template.
+
+![MongoDB Atlas Console](/images/blog/integrate-sql-nosql-vector-graph-or-any-database-into-your-appwrite-project/mongodb.png)
+
+## Redis Labs
+
+Redis is a widely used key-value database, most often serving as a cache layer in front of another database. It’s faster than the database behind it since Redis stores data in memory.
```

Comment:
```suggestion
Redis is a widely used key-value database, most often serving as a cache layer in front of another database. It’s faster than the underlying database since Redis stores data in memory.
```